pontoon/base/models.py

```diff
@@ -1926,6 +1940,11 @@ class Translation(DirtyFieldsMixin, models.Model):
     unapproved_date = models.DateTimeField(null=True, blank=True)
     fuzzy = models.BooleanField(default=False)
 
+    # Field contains an concatenated state of entity for faster search lookups.
+    # Due the nature of sql queries, it's faster to perform `icontains` filter the same table
+    # instead of OR condition for The Entity and The Translation class joined together.
```

Comment:
- **a** concatenated
- of **the** entity
- Due **to** the
- **on** the same table
- instead of -> **than**